code/game/objects/items/weapons/twohanded.dm

```diff
@@ -248,6 +248,59 @@
 	sharp = IS_SHARP_ITEM_SIMPLE
 	hitsound = 'sound/weapons/bladeslice.ogg'
 	attack_verb = list("attacked", "stabbed", "jabbed", "torn", "gored")
+	var/current_angle = 45	//Based on what direction the tip of the spear is pointed at in the sprite; maybe someone makes a spear that points northwest
+
+/obj/item/weapon/twohanded/spear/throw_at(atom/target, range, speed, thrower, spin, flying)
+	spin = FALSE
+	//Find the angle the spear is to be thrown at, then rotate it based on that angle at 45 degree increments
+	var/rotation_value = (Get_Angle(thrower, get_turf(target)))
+	switch(rotation_value)
+		if(337.5 to 360, 0 to 22.5)
+			rotation_value = 0 - current_angle
+		if(22.5 to 67.5)
+			rotation_value = 45 - current_angle
+		if(67.5 to 112.5)
+			rotation_value = 90 - current_angle
+		if(112.5 to 157.5)
+			rotation_value = 135 - current_angle
+		if(157.5 to 202.5)
+			rotation_value = 180 - current_angle
+		if(202.5 to 247.5)
+			rotation_value = 225 - current_angle
+		if(247.5 to 292.5)
+			rotation_value = 270 - current_angle
+		if(292.5 to 337.5)
+			rotation_value = 315 - current_angle
+	current_angle += rotation_value
+	var/matrix/rotate_me = matrix()
+	rotate_me.Turn(rotation_value)
+	src.transform = rotate_me
```

Comment:
Src usage

Reply:
Because it's only meant to reference itself? Parent proc does it as well, and so do the BYOND docs for doing transformations. Proc happens on the source.

Reply:
I haven't used transform much, but I don't see why src would be needed. transform is used on its own in various places in the code as is.

Reply:
using src.var unless var and an argument share a name is bad practice and we dont allow it

Reply:
It was what was shown in the BYOND reference for transformations, thought it was needed but makes sense now.